### DISCUSSION TITLE
Passing Data between Activities

### DISCUSSION THREAD

**Discussion by UserNameIsABC** [elsa-workflows/elsa-core#1356] [Q&A]
If I want to input a value (stream or byte data) from one activity to another,
what kind of source code can I use to achieve this?

The following image is attached as an example.

![activity_moji](https://user-images.githubusercontent.com/88302322/128444648-fff28c30-5743-4488-b330-f729c8c081cb.jpg)

I would like to input the value that is output by the activity "ReadFile" into the activity "HTTP Response".

There is a similar question in another post.
#1344
In your answer, you mentioned that you are using three activities to email stream data.
Is this something that will be supported in the future?

The sample you gave me ([elsa-guides/src/ElsaDemos.DocumentManagement/](https://github.com/elsa-workflows/elsa-guides/tree/master/src/ElsaDemos.DocumentManagement)) was not a sample of data transfer between three activities, so I asked again.

**Comment by mickdelaney** [comment]
From the docs here:

https://elsa-workflows.github.io/elsa-core/docs/next/guides/guides-blocking-activities#real-files

As it turns out, fixing this is easy. We can configure where to save activity input & output on a per-property basis using a pluggable storage provider mechanism called workflow storage providers.

Out of the box, Elsa ships with three such providers:

Workflow Instance Storage (the default)
Blob Storage
Transient Storage

**Comment by sfmskywalker** [comment]
@UserNameIsABC 

> In your answer, you mentioned that you are using three activities to email stream data.

I think I mentioned using three workflows for demo purposes. To email a stream as an attachment, you only really just need the Send Email activity and a way to provide the actual stream (which could come from a previous activity).

The documentation linked to by @mickdelaney should help. And [this blog post](https://medium.com/codex/part-6-of-building-workflow-driven-net-applications-with-elsa-2-9b3167c612dd) shows you how you might load a file from one activity and send it to the next activity to compress it into a zip stream. The same principle applies: one activity provides a stream as an output, and another activity is configured to do something with it (zip it, email it, write it to the response stream, send it to block chain, etc.).

**Reply by UserNameIsABC** [reply]
Sorry for the late reply.

I took the advice of both of you and implemented the [Writing Blocking Activities](https://elsa-workflows.github.io/elsa-core/docs/next/guides/guides-blocking-activities) page until the end.
Thanks to you, I was able to understand how the stream data that I "output" is received by other activities.

Also, I was able to display the stream data on the homepage.

Thank you very much for your detailed instruction.
I may have more questions about how to use the workflow in the future.
In that case, I would appreciate it if you could help me out again.

Thank you very much.
